src/Protocol/Diaspora/Repository/DiasporaContact.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Protocol\Diaspora\Repository;
+
+use Friendica\BaseRepository;
+use Friendica\Core\System;
+use Friendica\Database\Database;
+use Friendica\Database\Definition\DbaDefinition;
+use Friendica\Model\APContact;
+use Friendica\Model\Contact;
+use Friendica\Model\Item;
+use Friendica\Model\ItemURI;
+use Friendica\Network\HTTPException;
+use Friendica\Protocol\Diaspora\Entity;
+use Friendica\Protocol\Diaspora\Factory;
+use Friendica\Util\DateTimeFormat;
+use GuzzleHttp\Psr7\Uri;
+use Psr\Http\Message\UriInterface;
+use Psr\Log\LoggerInterface;
+
+class DiasporaContact extends BaseRepository
+{
+	const ALWAYS_UPDATE                 = true;
+	const NEVER_UPDATE                  = false;
+	const UPDATE_IF_MISSING_OR_OUTDATED = null;
+
+	protected static $table_name = 'diaspora-contact-view';
+
+	/**
+	 * @var Factory\DiasporaContact
+	 */
+	protected $factory;
+	/**
+	 * @var DbaDefinition
+	 */
+	private $definition;
+
+	public function __construct(DbaDefinition $definition, Database $database, LoggerInterface $logger, Factory\DiasporaContact $factory)
+	{
+		parent::__construct($database, $logger, $factory);
+
+		$this->definition = $definition;
+	}
+
+	/**
+	 * @param array $condition
+	 * @param array $params
+	 * @return Entity\DiasporaContact
+	 * @throws HTTPException\NotFoundException
+	 */
+	public function selectOne(array $condition, array $params = []): Entity\DiasporaContact
+	{
+		return parent::_selectOne($condition, $params);
+	}
+
+	/**
+	 * @param int $uriId
+	 * @return Entity\DiasporaContact
+	 * @throws HTTPException\NotFoundException
+	 */
+	public function selectOneByUriId(int $uriId): Entity\DiasporaContact
+	{
+		return $this->selectOne(['uri-id' => $uriId]);
+	}
+
+	/**
+	 * @param UriInterface $uri
+	 * @return Entity\DiasporaContact
+	 * @throws HTTPException\NotFoundException
+	 */
+	public function selectOneByUri(UriInterface $uri): Entity\DiasporaContact
+	{
+		$uriId = ItemURI::getIdByURI($uri, false);
+
+		return $this->selectOne(['`uri-id` = ? OR `alias` = ? OR `addr` = ?', $uriId, (string)$uri, (string)$uri]);
```

Comment:
Using `or` is bad for the performance because when performing the database query, the optimizer will have to pick one index. It's faster to perform separate requests. Also: why are you fetching the `uri-id` here? It is faster just to search for the `url` on the view.

Reply:
At some point I realized that the `dcontact` table didn't have the `url` field anymore so I scrambled to replace the references. Of course I forgot about the view at that time.